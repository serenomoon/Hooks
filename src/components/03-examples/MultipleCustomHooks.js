import React from 'react'
import { useCounter } from '../../hooks/useCounter';
import { useFetch } from '../../hooks/useFetch'

export const MultipleCustomHooks = () => {

    const {counter, increment} = useCounter(1);

    const {loading, data} = useFetch(`https://www.breakingbadapi.com/api/quotes/${counter}`);

    const {author, quote} = !!data && data[0];
//    console.log(state);
  return (
    
    <>
        <h1> Breaking Bad Quotes </h1>
        <hr />

        {
            loading ?
            (<div className='alert alert-info text-center'>
                Loading...
            </div>)
            :
            (<><blockquote className='blockquote text-right'>
                <p className='mb-0'>{quote}</p>
                <br></br>
                <footer className='blockquote-footer'>{author}</footer>
            </blockquote>

            <button 
            className='btn btn-primary'
            onClick={increment}
            >Next Quote</button>
            </>)
        }
    </>

  )
}
