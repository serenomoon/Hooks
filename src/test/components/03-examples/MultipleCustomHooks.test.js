import React from 'react';
import { shallow } from 'enzyme';
import { MultipleCustomHooks } from '../../../components/03-examples/MultipleCustomHooks';
import { useCounter } from '../../../hooks/useCounter';
import { useFetch } from '../../../hooks/useFetch';

jest.mock('../../../hooks/useCounter');
jest.mock('../../../hooks/useFetch');

describe('Test "MultipleCustomHooks.js"', () => {
    
    beforeEach( () =>  {
 
        useCounter.mockReturnValue({
            counter: 10,
            increment: () => {}
        });
 
    });

    test('should show correct snapshot', () => {
       
        useFetch.mockReturnValue({
            data: null,
            loading:true,
            error:null
        });

        const wrapper = shallow( <MultipleCustomHooks /> );

        expect( wrapper ).toMatchSnapshot();

    });

    test('should return correct info', () => {
       
        useFetch.mockReturnValue({
            data: [{
                author: 'Saulo',
                quote: 'Hola Mundo'
            }],
            loading: false,
            error: null
        });

        const wrapper = shallow( <MultipleCustomHooks /> );

        expect( wrapper.find('.alert').exists() ).toBe( false );
        expect( wrapper.find('.mb-0').text().trim() ).toBe( 'Hola Mundo' );
        expect( wrapper.find('footer').text().trim() ).toBe( 'Saulo' );

    });

});