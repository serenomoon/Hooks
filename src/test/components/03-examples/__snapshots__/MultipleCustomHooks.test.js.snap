// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test "MultipleCustomHooks.js" should show correct snapshot 1`] = `
<Fragment>
  <h1>
     Breaking Bad Quotes 
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    Loading...
  </div>
</Fragment>
`;
